res/css/components/views/settings/devices/_DeviceDetailHeading.pcss

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+.mx_DeviceDetailHeading {
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    align-items: center;
+}
+
+.mx_DeviceDetailHeading_renameCta {
+    flex-shrink: 0;
+}
+
+.mx_DeviceDetailHeading_renameForm {
+    display: grid;
```

Comment:
We could use `flex` here. With 100 % width this behaves like a flex layout.
(Or I just don't know a feature of `grid` :laughing: )